contracts/interfaces/IPriceChecker.sol

```diff
@@ -2,7 +2,7 @@
 pragma solidity ^0.8.13;
 
 interface IPriceChecker {
-    function getExpectedOut(uint256 amount, address tokenFrom, address tokenTo, bytes calldata)
+    function getExpectedOut(uint256 amount, address tokenFrom, address tokenTo, uint256 margin)
```

Comment:
The `PriceChecker` title sounds too unspecified to me. From the name, it looks more like a contract with assert checks. But what the contract actually does is a price conversion: takes the price in token A and returns it nominated in TokenB. I would change the name of the interface to `TokensConverter` or similar. 

Also, I would remove the `margin` from the interface and apply it already in the order contract. Or at least would change the name to something like a `discount`